### PR TITLE
Add smfi command for setting fan curve

### PR DESCRIPTION
**Pull request by mkopec**
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

### CODE REVIEW

File: src/board/system76/common/dgpu.c [hunk 1]
```diff
@@ -56,6 +56,22 @@ static struct Fan __code FAN = {
...
 };
 
+int16_t dgpu_set_fan_curve(uint8_t count, struct FanPoint * points) {
+    if (count != FAN.points_size) {
+        TRACE("DGPU: Incorrect number of fan points: %d, expected %d\n",
+            count, FAN.points_size);
+        return -1;
+    }
+
+    // TODO: check for curve validity? Or assume BIOS / user are sane?
```
Comment (miczyg1) [inline src/board/system76/common/dgpu.c:66]: Yes, please do the check...

File: src/board/system76/common/peci.c [hunk 2]
```diff
@@ -56,6 +56,22 @@ static struct Fan __code FAN = {
...
 };
 
+int16_t peci_set_fan_curve(uint8_t count, struct FanPoint * points) {
+    if (count != FAN.points_size) {
+        TRACE("PECI: Incorrect number of fan points: %d, expected %d\n",
+            count, FAN.points_size);
+        return -1;
+    }
+
+    // TODO: check for curve validity? Or assume BIOS / user are sane?
```
Comment (miczyg1) [inline src/board/system76/common/peci.c:66]: Yes please do the check

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: src/board/system76/common/dgpu.c (hunk 1 above)
Comment (mkopec) [reply src/board/system76/common/dgpu.c]: Added a check for curve sanity in smfi.c - comment removed

File: src/board/system76/common/peci.c (hunk 2 above)
Comment (mkopec) [reply src/board/system76/common/peci.c]: Added a check for curve sanity in smfi.c - comment removed

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**